src/videojs.markers.js

```diff
@@ -393,13 +392,16 @@ function registerVideoJsMarkersPlugin(options) {
     }
     onTimeUpdate();
     player.on("timeupdate", onTimeUpdate);
-    player.off("loadedmetadata");
+    // player.off("loadedmetadata");
   }
 
   // setup the plugin after we loaded video's meta data
-  player.on("loadedmetadata", function() {
+  // player.on("loadedmetadata", function() {
+  //   initialize();
+  // });
+  player.ready(function(){
```

Comment:
Nice! Does `.ready` guarantee the duration information has been loaded? My guess is that @spchuang had this listening to `loadedmetadata` for that reason.